### PR TITLE
Migration path check when registration sharing is enabled

### PR DESCRIPTION
**Pull request by ikapelyukhin**

### CODE REVIEW

**Comment by ikapelyukhin** [comment]
@rjschwei The code now should be doing pretty much the same thing it was doing before offline migrations were implemented:

https://github.com/SUSE/smt/blob/39bf7dfb1e108383a947242345c11767e3d601d6/www/perl-lib/SMT/Rest/SCCAPIv4.pm#L299-L311

https://github.com/SUSE/smt/blob/39bf7dfb1e108383a947242345c11767e3d601d6/www/perl-lib/SMT/Rest/SCCAPIv4.pm#L332-L336

**Review by rjschwei — APPROVED** [review]

**Comment by ikapelyukhin** [comment]
@rjschwei Improved logic, as we've discussed.

**Review by rjschwei — APPROVED** [review]
LGTM, thanks

File: www/perl-lib/SMT/Rest/SCCAPIv4.pm [hunk 1]
```diff
@@ -335,6 +335,24 @@ sub update_product
...
+            # disallow base product change
+            return (
+                Apache2::Const::HTTP_UNPROCESSABLE_ENTITY,
+                "No installed product with requested migration target product found"
+            ) unless (SMT::Utils::checkMigrationPath($self->dbh(), $req_pdid, $regs));
+        } else {
+            # require extension dependency to be activated
+            return (
+                Apache2::Const::HTTP_UNPROCESSABLE_ENTITY,
+                "Product dependency activation not found"
```
Comment (wstephenson) [inline www/perl-lib/SMT/Rest/SCCAPIv4.pm]: Could this error be improved?

Something like "No activation for dependency of product $xyz"?

Comment (ikapelyukhin) [reply www/perl-lib/SMT/Rest/SCCAPIv4.pm]: @wstephenson Well, you are never supposed to encounter it, unless you are trying to upgrade to something you aren't allowed to upgrade to.

Comment (wstephenson) [reply www/perl-lib/SMT/Rest/SCCAPIv4.pm]: That doesn't mean it should be incomprehensible when you do make that mistake.

Comment (wstephenson) [reply www/perl-lib/SMT/Rest/SCCAPIv4.pm]: Otherwise, lgtm

**Review by wstephenson — APPROVED** [review]

**Pull request merged**